src/crypto-providers/ledgerCryptoProvider.ts

```diff
@@ -0,0 +1,363 @@
+import { Errors } from '../errors'
+import {
+  TxByronWitness,
+  TxShelleyWitness,
+  TxSigned,
+  XPubKey,
+} from '../transaction/transaction'
+import {
+  _Input,
+  _Output,
+  SignedTxCborHex,
+  _TxAux,
+  _ByronWitness,
+  _ShelleyWitness,
+  TxWitnessByron,
+  TxWitnessShelley,
+  XPubKeyHex,
+  _Certificate,
+  TxCertificateKeys,
+  _Withdrawal,
+  _StakepoolRegistrationCert,
+  _DelegationCert,
+  _StakingKeyDeregistrationCert,
+  _StakingKeyRegistrationCert,
+  _PoolRelay,
+  _SingleHostIPRelay,
+  _MultiHostNameRelay,
+  _SingleHostNameRelay,
+  TxRelayTypes,
+  TxWitnessKeys,
+} from '../transaction/types'
+import {
+  Address,
+  BIP32Path,
+  HwSigningData,
+  Network,
+} from '../types'
+import {
+  LedgerCertificate,
+  LedgerInput,
+  LedgerMultiHostNameRelay,
+  LedgerOutput,
+  LedgerPoolOwnerParams,
+  LedgerPoolParams,
+  LedgerRelayParams,
+  LedgerSingleHostIPRelay,
+  LedgerSingleHostNameRelay,
+  LedgerWithdrawal,
+  LedgerWitness,
+} from './ledgerTypes'
+import { CryptoProvider, _AddressParameters } from './types'
+import {
+  filterSigningFiles,
+  findSigningPath,
+  getChangeAddress,
+  getSigningPath,
+  isShelleyPath,
+  getAddressAttributes,
+  rewardAddressToPubKeyHash,
+  ipv4ToString,
+  ipv6ToString,
+} from './util'
+
+const TransportNodeHid = require('@ledgerhq/hw-transport-node-hid').default
+const Ledger = require('@cardano-foundation/ledgerjs-hw-app-cardano').default
+
+export const LedgerCryptoProvider: () => Promise<CryptoProvider> = async () => {
+  const transport = await TransportNodeHid.create()
+  const ledger = new Ledger(transport)
+
+  const getVersion = async (): Promise<string> => {
+    const { major, minor, patch } = await ledger.getVersion()
+    return `Ledger app version ${major}.${minor}.${patch}`
+  }
+
+  const showAddress = async (
+    paymentPath: BIP32Path, stakingPath: BIP32Path, address: Address,
+  ): Promise<void> => {
+    try {
+      const { addressType, networkId } = getAddressAttributes(address)
```

Comment:
will this work for byron addresses?

Reply:
It didn't, fixed. Also, it might be a good idea to add `--mainnet | --testnet-magic NATURAL` as a parameter to the command.